llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -295,15 +296,95 @@ saveResultModules(std::vector<std::unique_ptr<Module>> &ResModules) {
   return Res;
 }
 
+// Each fallback device library corresponds to one bit in "require mask" which
+// is an unsigned int32. getDeviceLibBit checks which fallback device library
+// is required for FuncName and returns the corresponding bit. The corresponding
+// mask for each fallback device library is:
+// fallback-cassert:      0x1
```

Comment:
The library identification in program_manager.hpp and post link tool should match, so I suggest:
1) Add a cross-reference between sycl-post-link.cpp and program_manager.hpp and a comment that the library ids should match.
2) introduce `enum DeviceLibExt` exactly matching what's in program_manager.hpp here, so that grep over the sources gives all related occurrences.
3) Instead of individual definitions like `static constexpr uint32_t DeviceLib*` here use bit shift operations to translate e.g. `cl_intel_devicelib_math_fp64` to the corresponding bitmask's bit and back. Then we'll have to maintain single enum (in two places) rather than two inter-related.